core/finality-grandpa/src/lib.rs

```diff
@@ -523,20 +523,24 @@ pub fn run_grandpa_voter<B, E, Block: BlockT<Hash=H256>, N, RA, SC, X>(
 
 	register_finality_tracker_inherent_data_provider(client.clone(), &inherent_data_providers)?;
 
+	let conf = config.clone();
 	let telemetry_task = if let Some(telemetry_on_connect) = telemetry_on_connect {
 		let authorities = persistent_data.authority_set.clone();
 		let events = telemetry_on_connect
 			.for_each(move |_| {
+				let maybe_authority_id = authority_id(&authorities.current_authorities(), &conf.keystore)
```

Comment:
Hmm, when we rotate the session and the key changes, this will not notify the telemetry without a disconnect.

Reply:
Updated the PR to address this (after talking to @andresilva about a fitting place in the code).